### PR TITLE
fix isValidDim for block arg case

### PR DESCRIPTION
**Pull request by bondhugula**
Signed-off-by: Uday Bondhugula <uday@polymagelabs.com>

### CODE REVIEW

File: test/Transforms/slicing-utils.mlir [hunk 1]
```diff
@@ -222,13 +222,12 @@ func @slicing_test() {
 // FWDBWD-LABEL: slicing_test_2
 func @slicing_test_2() {
   %c0 = constant 0 : index
-  %c1 = constant 1 : index
   %c2 = constant 2 : index
   %c16 = constant 16 : index
-  loop.for %i0 = %c0 to %c16 step %c1 {
```
Comment (ftynse) [inline test/Transforms/slicing-utils.mlir]: This change doesn't look relevant to the commit.

Comment (bondhugula) [reply test/Transforms/slicing-utils.mlir]: It is :-) With a loop.for %i0 isn't a valid dimensional identifier for anything inside (only affine.for IVs are).

**Comment by bondhugula** [comment]
> Could you please describe what is the problem that you are fixing in the commit message? Thanks!

I was thinking the test case was sufficient - but have updated the commit message. Thanks!

File: test/Transforms/slicing-utils.mlir (hunk 1 above)
Comment (ftynse) [reply test/Transforms/slicing-utils.mlir]: My bad, I missed it was used in the map in the nested `affine.for`.

**Comment by ftynse** [comment]
> > Could you please describe what is the problem that you are fixing in the commit message? Thanks!
> 
> I was thinking the test case was sufficient - but have updated the commit message. Thanks!

I'm nitpicking, but as the scale of the project grows, I'm expecting some people following the development will read commit messages but will not take the time to look into the actual change. Thanks!

**Comment by bondhugula** [comment]
> > > Could you please describe what is the problem that you are fixing in the commit message? Thanks!
> > 
> > 
> > I was thinking the test case was sufficient - but have updated the commit message. Thanks!
> 
> I'm nitpicking, but as the scale of the project grows, I'm expecting some people following the development will read commit messages but will not take the time to look into the actual change. Thanks!

Yes, you are right, I agree.
